src/ImmersedBoundaries/partial_cell_immersed_boundaries.jl

```diff
@@ -0,0 +1,98 @@
+using Adapt
+using CUDA: CuArray
+using Oceananigans.Fields: fill_halo_regions!
+using Oceananigans.Architectures: arch_array
+using Printf
+
+import Oceananigans.TurbulenceClosures: ivd_upper_diagonal,
```

Comment:
Are you changing the `ivd_upper_diagonal`?

Reply:
I don't think so --- that's a mistake!

Reply:
I just removed it since it indeed wasn't being used